imbi/automations/consul.py

```diff
@@ -0,0 +1,259 @@
+                                  method='GET',
+                                  raise_error=False)
+        if response.ok:
+            for policy in response.body.get('Policies', []):
+                self.logger.info('removing policy %r during rollback',
+                                 policy['ID'])
+                await self.api(self.api_url / 'acl' / 'policy' / policy['ID'],
+                               method='DELETE',
+                               raise_error=False)
+            self.logger.info('removing token %r during rollback', accessor_id)
+            await self.api(self.api_url / 'acl' / 'token' / accessor_id,
+                           method='DELETE',
+                           raise_error=False)
+
+    async def api(self,
+                  path: str | yarl.URL,
+                  *,
+                  method: str,
+                  raise_error=True,
+                  **kwargs):
+        if isinstance(path, yarl.URL):
+            url = path
+        else:
+            url = self.api_url / path.lstrip('/')
+
+        request_headers = self.headers.copy()
+        request_headers.update(kwargs.pop('request_headers', {}))
+        response = await super().http_fetch(
+            str(url),
+            method=method,
+            request_headers=request_headers,
+            user_agent=f'imbi/{version} (ConsulClient)',
+            dont_retry={500},
+            **kwargs)
+        if raise_error and not response.ok:
+            raise errors.InternalServerError('%s %s failed: %s',
+                                             method,
+                                             url,
+                                             response.code,
+                                             title='Consul API Failure')
+        return response
+
+
+class ConsulCreateTokenAutomation(base.Automation):
+    def __init__(self, application, project_id, current_user, db) -> None:
+        super().__init__(application, current_user, db)
+        self.imbi_project_id = project_id
+        self.project = None
+        self.settings = self.automation_settings['consul']
+        self.enabled = bool(self.settings.get('environments'))
+
+    async def prepare(self) -> list[str]:
+        if not self.enabled:
+            self.logger.warning(
+                'consul token creation disabled: no environment tokens '
+                'configured')
+            return self.errors
+
+        project = await self._get_project(self.imbi_project_id)
+        if project is None:
+            self._add_error('Project id {} does not exist',
+                            self.imbi_project_id)
+        elif project.project_type.name not in self.settings['enabled_for']:
+            self.logger.warning(
+                'disabling consul automation since project type %r not '
+                'found in %r', project.project_type.name,
+                self.settings['enabled_for'])
+            self.enabled = False
```

Comment:
Any reason why this isn't also considered an error case? Instead it looks like it will do nothing silently and return a 204.